### PR TITLE
Update LIP 0045 Genesis State Finalization

### PR DESCRIPTION
**Pull request by gkoumout**
This PR resolves [this issue](https://github.com/LiskHQ/lips/issues/320)

### CODE REVIEW

**Review by ricott1 — APPROVED** [review]

**Review by ishantiw — APPROVED** [review]
Related SDK issue https://github.com/LiskHQ/lisk-sdk/issues/8338

**Review by ricott1 — APPROVED** [review]

File: proposals/lip-0045.md [hunk 1]
```diff
@@ -1549,7 +1549,7 @@ On the mainchain, the following checks are performed:
   * check that `totalWeight//3 + 1 <= certificateThreshold <= totalWeight`, where `//` indicates integer division;
   * check that the corresponding `validatorsHash` stored in `chainInfo.chainData.lastCertificate.validatorsHash` matches with the value computed from `activeValidators` and `certificateThreshold`.
 * For each entry `chainInfo` in `chainInfos`, `chainInfo.chainData.status == CHAIN_STATUS_TERMINATED` if and only if a corresponding entry (i.e., with `chainID == chainInfo.chainID`) exists in `terminatedStateAccounts`.
-* Each entry `stateAccount` in `terminatedStateAccounts` has a unique `stateAccount.chainID` and `terminatedStateAccounts` is ordered lexicographically by `stateAccount.chainID`.
+* Each entry `stateAccount` in `terminatedStateAccounts` has a unique `stateAccount.chainID` and `terminatedStateAccounts` is ordered lexicographically by `stateAccount.chainID`. Furthermore for each entry it holds `stateAccount.chainID != ownChainAccount.chainID`.
```
Comment (ricott1) [inline proposals/lip-0045.md:1552]: I would check this as we do above:

* `stateAccount.chainID != getMainchainID()`;
* `stateAccount.chainId[0] == getMainchainID()[0]`.

Comment (gkoumout) [reply proposals/lip-0045.md:1552]: Ok makes sense. I guess also add this check for first byte in the corresponding sidechain check (line 1586)?

Comment (gkoumout) [reply proposals/lip-0045.md]: Added.

File: proposals/lip-0045.md [hunk 2]
```diff
@@ -1583,7 +1583,7 @@ If `chainInfos` is not empty, then check that:
   * let `totalWeight` be the sum of the `bftWeight` property of every element in `activeValidators`. Then `totalWeight` has to be less than or equal to `MAX_UINT64`;
   * check that `totalWeight//3 + 1 <= certificateThreshold <= totalWeight`, where `//` indicates integer division;
   * check that the corresponding `validatorsHash` stored in `mainchainInfo.chainData.lastCertificate.validatorsHash` matches with the value computed from `activeValidators` and `certificateThreshold`.
-* Each entry `stateAccount` in `terminatedStateAccounts` has a unique `stateAccount.chainID` and `terminatedStateAccounts` is ordered lexicographically by `stateAccount.chainID`. Furthermore for each entry it holds `stateAccount.chainID != getMainchainID()` and `stateAccount.chainID != ownChainAccount.chainID`.
+* Each entry `stateAccount` in `terminatedStateAccounts` has a unique `stateAccount.chainID` and `terminatedStateAccounts` is ordered lexicographically by `stateAccount.chainID`. Furthermore for each entry it holds `stateAccount.chainID != getMainchainID()`, `stateAccount.chainID != ownChainAccount.chainID` and `stateAccount.chainId[0] == getMainchainID()[0]`.
```
Comment (ricott1) [inline proposals/lip-0045.md:1586]: Is `ownChainAccount` actually defined already at this point? Shouldn't we check against `OWN_CHAIN_ID` and initialize the `ownChainAccount` in the genesis state finalization?

**Review by ricott1 — CHANGES_REQUESTED** [review]
Added one comment about `ownChainAccount` initialization

File: proposals/lip-0045.md (hunk 2 above)
Comment (gkoumout) [reply proposals/lip-0045.md]: Good point. Changed the check.

**Review by ricott1 — APPROVED** [review]

**Review by janhack — APPROVED** [review]
Editorial approval

**Pull request merged**